src/core/decrypt/utils/key_session_record.ts

```diff
@@ -148,8 +148,8 @@ export default class KeySessionRecord {
       if (this._keyIds !== null && areAllKeyIdsContainedIn(keyIds, this._keyIds)) {
         return true;
       }
-      if (this._initializationData.keyIds !== undefined) {
-        return areAllKeyIdsContainedIn(keyIds, this._initializationData.keyIds);
+      if (this._initializationData.keyIds !== undefined && areAllKeyIdsContainedIn(keyIds, this._initializationData.keyIds)) {
```

Comment:
To me, if a wanted key id is not linked to a session, then the license has to be loaded, else we would be breaking legitimate use cases.